src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -243,66 +243,72 @@ struct AccessControlEntryCodec
                 break;
             }
             case to_underlying(Fields::kSubjects): {
-                TLV::TLVType subjectsContainer;
-                VerifyOrReturnError(TLV::kTLVType_Array == aReader.GetType(), CHIP_ERROR_WRONG_TLV_TYPE);
-                ReturnErrorOnFailure(aReader.EnterContainer(subjectsContainer));
-                while ((err = aReader.Next()) == CHIP_NO_ERROR)
+                if (aReader.GetType() != TLV::kTLVType_Null)
                 {
-                    NodeId subject = kUndefinedNodeId;
-                    ReturnErrorOnFailure(DataModel::Decode(aReader, subject));
-                    ReturnErrorOnFailure(entry.AddSubject(nullptr, subject));
+                    TLV::TLVType subjectsContainer;
+                    VerifyOrReturnError(TLV::kTLVType_Array == aReader.GetType(), CHIP_ERROR_WRONG_TLV_TYPE);
+                    ReturnErrorOnFailure(aReader.EnterContainer(subjectsContainer));
+                    while ((err = aReader.Next()) == CHIP_NO_ERROR)
+                    {
+                        NodeId subject = kUndefinedNodeId;
+                        ReturnErrorOnFailure(DataModel::Decode(aReader, subject));
+                        ReturnErrorOnFailure(entry.AddSubject(nullptr, subject));
+                    }
+                    VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
+                    ReturnErrorOnFailure(aReader.ExitContainer(subjectsContainer));
                 }
-                VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
-                ReturnErrorOnFailure(aReader.ExitContainer(subjectsContainer));
                 break;
             }
             case to_underlying(Fields::kTargets): {
-                TLV::TLVType targetsContainer;
```

Comment:
Similar comments here, though I guess there's at least some argument for not decoding into an intermediate Structs::Target...

Reply:
Avoids decoding an intermediate list.

Reply:
It's possible to make a sub-codec for the Target as well, so it can reuse more of the machinery without reusing the generated struct types. But that is not highest on my TODO list. The purpose of this change is to make the REPL work properly without having to specifying empty lists whereever they could be omitted (which then is null).

Reply:
So... it looks like this is pre-existing, but the code here is not spec-compliant.  The auto-generated structs were updated in https://github.com/project-chip/connectedhomeip/pull/13072 bit because this code is hand-rolled it missed the boat.

I'm going to resolve this conversation, but this code really needs to be fixed.